App.js

```diff
@@ -100,40 +98,35 @@ const CommonHome = () => {
 
 const App = ({userStore}) => {
   const [onboarded, setOnboarded] = useState();
-  useEffect(() => {
-    const loadUser = async () => {
-      try {
-        if (!userStore.userInfo) {
-          console.log('LOAD USER 1');
-          const googleSignedInUser = await authService.getGoogleSignedInUser();
-          // Signed In Mode
-          if (googleSignedInUser) {
-            userStore.setIsLoading(true);
-            const appUser = await firebaseService.getUserById(
-              googleSignedInUser.user.id,
-            );
 
-            const allUserInfo = {
-              ...googleSignedInUser.user,
-              ...appUser,
-            };
+  const onAuthStateChanged = async user => {
+    try {
+      userStore.setIsLoading(true);
+      if (user) {
+        const appUser = await FirebaseService.getInstance().getUserById(
+          user.uid,
+        );
 
-            const filteredUser = filterObjectByKeys(
-              allUserInfo,
-              userInfoFields,
-            );
-            userStore.setSignedInUser(filteredUser);
-            userStore.setIsLoading(false);
-          }
-          // Anonymous mode
-          else {
-            console.log('Anonymous user');
-          }
-        }
-      } catch (error) {
-        console.log('ERRROR', error);
+        const allUserInfo = {
+          ...user._user,
+          ...appUser,
+        };
+
+        const filteredUser = filterObjectByKeys(allUserInfo, userInfoFields);
+
+        userStore.setSignedInUser(filteredUser);
+        userStore.setIsLoading(false);
+      } else {
+        userStore.setSignedInUser(null);
       }
-    };
+      userStore.setIsLoading(false);
+    } catch (error) {
+      console.log('ERRROR', error);
```

Comment:
This is not good: we have an error, and we just log it to the console, but we leave the user in the "loading..." state without any feedback.

Either just remove the try... catch... here, or, perhaps better, show the user an error here (using haos error messaging thing)

Reply:
This is also removed code.